R/get_v1_data.R

```diff
@@ -0,0 +1,243 @@
+  if(length(xml_files_list) == 0) {
+    if(isFALSE(quiet)) message("No data links xml files found, getting latest data links xml")
+    latest_data_links_xml_path <- spod_get_latest_v1_file_list(data_dir = data_dir)
+  } else {
+    latest_data_links_xml_path <- utils::tail(xml_files_list, 1)
+  }
+
+  # Check if the XML file is 1 day old or older from its name
+  file_date <- stringr::str_extract(latest_data_links_xml_path, "[0-9]{4}-[0-9]{2}-[0-9]{2}")
+
+  if (file_date < format(Sys.Date(), format = "%Y-%m-%d")) {
+    if(isFALSE(quiet)) message("File list xml is 1 day old or older, getting latest data links xml")
+    latest_data_links_xml_path <- spod_get_latest_v1_file_list(data_dir = data_dir)
+  } else {
+    if(isFALSE(quiet)) message("Using existing data links xml: ", latest_data_links_xml_path)
+  }
+
+  if (length(latest_data_links_xml_path) == 0) {
+    if(isFALSE(quiet)) message("Getting latest data links xml")
+    latest_data_links_xml_path <- spod_get_latest_v1_file_list(data_dir = data_dir)
+  }
+
+  x_xml <- xml2::read_xml(latest_data_links_xml_path)
+
+  files_table <- tibble::tibble(
+    target_url = xml2::xml_find_all(x = x_xml, xpath = "//link") |> xml2::xml_text(),
+    pub_date = xml2::xml_find_all(x = x_xml, xpath = "//pubDate") |> xml2::xml_text()
+  )
+
+  files_table$pub_ts <- lubridate::dmy_hms(files_table$pub_date)
+  files_table$file_extension <- tools::file_ext(files_table$target_url)
+  files_table <- files_table[files_table$file_extension != "", ]
+  files_table$pub_date <- NULL
+
+  files_table$data_ym <- lubridate::ym(stringr::str_extract(files_table$target_url, "[0-9]{4}-[0-9]{2}"))
+  files_table$data_ymd <- lubridate::ymd(stringr::str_extract(files_table$target_url, "[0-9]{8}"))
+  # order by pub_ts
+  files_table <- files_table[order(files_table$pub_ts, decreasing = TRUE), ]
+  files_table$local_path <- file.path(
+    data_dir,
+    stringr::str_replace(files_table$target_url, ".*mitma.es/", spod_subfolder_raw_data_cache(ver = 1))
+  )
+
+  files_table$local_path <- stringr::str_replace_all(files_table$local_path, "\\/\\/\\/|\\/\\/", "/")
+
+  # change path for daily data files to be in hive-style format
+  files_table$local_path <- gsub("([0-9]{4})-([0-9]{2})\\/[0-9]{6}([0-9]{2})_", "year=\\1\\/month=\\2\\/day=\\3\\/", files_table$local_path)
+
+  # fix paths for files that are in '0000-referencia' folder
+  files_table$local_path <- gsub("0000-referencia\\/([0-9]{4})([0-9]{2})([0-9]{2})_", "year=\\1\\/month=\\2\\/day=\\3\\/", files_table$local_path)
+  
+  # replace 2 digit month with 1 digit month
+  files_table$local_path <- gsub("month=0([1-9])", "month=\\1", files_table$local_path)
+  
+  # replace 2 digit day with 1 digit day
+  files_table$local_path <- gsub("day=0([1-9])", "day=\\1", files_table$local_path)
+
+  # now check if any of local files exist
+  files_table$downloaded <- fs::file_exists(files_table$local_path)
+
+  return(files_table)
+}
+
+#' Retrieves the zones for v1 data
```

Comment:
Great to have zones v1, different from v2 I guess.

Reply:
Internally the code for v1 and v2 zones will be very different. We can have `spod_get_zones(ver = 1)`, `spod_get_zones(ver = 2)`, as well as `spod_get_zones_v1()` and `spod_get_zones_v2()`. I think wrapper functions won't hurt.